From fa27f3e27863e3f4376864b144dc7506de124cf6 Mon Sep 17 00:00:00 2001
From: Jason Jin <Jason.jin@freescale.com>
Date: Thu, 4 Aug 2011 09:59:54 +0800
Subject: [PATCH 43/52] workaround for zero page used on ColdFire platform.

There is some issue for Coldfire platform to use the
zero page for read when there is memory fault. This is
a workaround for this, try to make kernel to use normal
memory instead.

Signed-off-by: Jason Jin <Jason.jin@freescale.com>
---
 mm/memory.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

--- a/mm/memory.c
+++ b/mm/memory.c
@@ -2937,6 +2937,7 @@ static int do_anonymous_page(struct mm_s
 	if (check_stack_guard_page(vma, address) < 0)
 		return VM_FAULT_SIGBUS;
 
+#ifndef CONFIG_COLDFIRE
 	/* Use the zero-page for reads */
 	if (!(flags & FAULT_FLAG_WRITE)) {
 		entry = pte_mkspecial(pfn_pte(my_zero_pfn(address),
@@ -2946,6 +2947,7 @@ static int do_anonymous_page(struct mm_s
 			goto unlock;
 		goto setpte;
 	}
+#endif
 
 	/* Allocate our own private page. */
 	if (unlikely(anon_vma_prepare(vma)))
